### ISSUE TITLE
AMFIOCaps::GetFormatAt returns random AMF_SURFACE_FORMAT values

### ISSUE THREAD

**Issue by mfleisz** [GPUOpen-LibrariesAndSDKs/AMF#271]
Since I upgraded my display driver to version 21.10.1 `AMFIOCaps::GetFormatAt` returns random format values for output caps. 

Sample code:

```
	amf::AMFIOCapsPtr outputCaps;
	if (res == AMF_OK)
		res = encoderCaps->GetOutputCaps(&outputCaps);
	if (res == AMF_OK)
	{
		amf_uint32 curFormat;
		amf_uint32 numFormats = caps->GetNumOfFormats();
		for (curFormat = 0; curFormat < numFormats; ++curFormat)
		{
			bool isNative;
			amf::AMF_SURFACE_FORMAT gotFormat;
			if (caps->GetFormatAt(curFormat, &gotFormat, &isNative) == AMF_OK)
			{
				printf("%d\n", gotFormat);  // This prints a different value every time the app runs
			}
		}
	}
```

The encoder was created with the following code:

```
	/* create encoder */
	if (res == AMF_OK)
	{
		res = g_AMFFactory.GetFactory()->CreateComponent(ve->amf.context, AMFVideoEncoderVCE_AVC, &ve->amf.encoder);
	}

	/* setup and init encoder */
	if (res == AMF_OK && ve->amf.encoder)
	{
		res = ve->amf.encoder->SetProperty(AMF_VIDEO_ENCODER_USAGE, AMF_VIDEO_ENCODER_USAGE_LOW_LATENCY);
		if (res == AMF_OK)
			res =
			    ve->amf.encoder->SetProperty(AMF_VIDEO_ENCODER_QUALITY_PRESET, AMF_VIDEO_ENCODER_QUALITY_PRESET_SPEED);
		if (res == AMF_OK)
			res = ve->amf.encoder->SetProperty(AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD,
			                                   AMF_VIDEO_ENCODER_RATE_CONTROL_METHOD_CONSTANT_QP);
		if (res == AMF_OK)
			res = ve->amf.encoder->SetProperty(AMF_VIDEO_ENCODER_FRAMESIZE, ::AMFConstructSize(ve->width, ve->height));
		if (res == AMF_OK)
			res = ve->amf.encoder->SetProperty(AMF_VIDEO_ENCODER_FRAMERATE, ::AMFConstructRate(30, 1));
		if (res == AMF_OK)
			res = ve->amf.encoder->Init(amf::AMF_SURFACE_NV12, ve->width, ve->height);
	}
```

3D API: Direct3D
OS: Windows 10 Pro x64 21H1
Radeon Software: 21.10.1
GPU: AMD Radeon RX 5700 XT 50th Anniversary

Let me know if you need any more information.

**Comment by rhutsAMD** [comment]
This issue has been fixed. No actions are needed from your side, and we will notify you once the driver containing the fix is made publicly available.

**Comment by rhutsAMD** [comment]
The fix for the random surface format values in the output caps is now publicly released in the Radeon™ Software Adrenalin 21.11.3 driver and I confirmed it is working with this driver.

**Comment by mfleisz** [comment]
Just installed the latest driver and I can confirm it is fixed - thanks!
